Separated captcha logic from http handling

LanerCodire opened the pull request:
Separating captcha logic from http handling (reading request, creating response) allows reuse of captcha logic in other use cases, e.g. in an existing http handling infrastructure or securing a REST endpoint

amenk commented:
Thanks, good idea.
Can you change it to PSR-4 and composer based autoloading? This way it can be easier used in other projects.

amenk commented:
@dev-love Could we publish this later on https://packagist.org/ ? Do you have an account / namespace yet?

LanerCodire commented:
> Thanks, good idea. Can you change it to PSR-4 and composer based autoloading? This way it can be easier used in other projects.

I also thought about autoloading, but I see this as another PR then.

amenk commented:
I would try to avoid to change the class name convention twice ... so .class should be removed from the files and they should start with a capital letter?

LanerCodire commented:
> I would try to avoid to change the class name convention twice ... so .class should be removed from the files and they should start with a capital letter?

I changed the classes to use PSR-4 style and also provided a default vendor/autoload.php which may be replaced by composer's one. Moreover I changed the configuration (env.php) to be a class as well.

amenk commented on public/puzzle.php:
```diff
@@ -1,66 +1,19 @@
+use FriendlyCaptcha\Captcha;
+use FriendlyCaptcha\Polite;
+
+require_once 'vendor/autoload.php';
```
would just put that in the same folder - otherwise it might conflict with composer?

LanerCodire replied:
The "conflict" is intended. If you use composer, you just replace the autoload.php by composer's version and do not need to adopt the require statements.

amenk commented on public/classes/FriendlyCaptcha/Captcha.php:
```diff
@@ -0,0 +1,153 @@
+<?php
```
can we put them all in src/ and just keep puzzle.php and siteverify.php in public/?

LanerCodire replied:
Sure. Do you prefer "src/classes/FriendlyCaptcha/Captcha.php" or "src/FriendlyCaptcha/Captcha.php"?

amenk replied:
without classes/ .... and while we are on this, we should add another sub-namespace.

FriendlyCaptcha/Lite/Captcha.php ...

LanerCodire replied:
I moved the classes as requested

amenk commented:
Looks good to me. Thank you very much - @dev-love will have the final say

amenk commented:
thanks @LanerCodire
